pkg/exchange/coinbase/stream_test.go

```diff
@@ -49,3 +54,199 @@ func Test_SubCmdString(t *testing.T) {
+
+	// TODO: test Rfq message
+}
+
+func TestStreamFull(t *testing.T) {
+	c := make(chan bool)
+	productIDs := []string{"BTC-USD", "ETH-USD"}
+
+	t.Run("Run Full", func(t *testing.T) {
+		stream := getTestStreamOrSkip(t)
+		for _, productID := range productIDs {
+			stream.Subscribe("full", productID, types.SubscribeOptions{})
+		}
+		// TODO: test full order life cycle
+		// received -> open -> change* -> match? -> done
+		stream.OnReceivedMessage(func(m *ReceivedMessage) {
+			t.Log("get received message")
+			assert.NotNil(t, m)
+			c <- true
+		})
+		stream.OnOpenMessage(func(m *OpenMessage) {
+			t.Log("get open message")
+			assert.NotNil(t, m)
+			c <- true
+		})
+		stream.OnDoneMessage(func(m *DoneMessage) {
+			t.Log("get done message")
+			assert.NotNil(t, m)
+			c <- true
+		})
+		stream.OnMatchMessage(func(m *MatchMessage) {
+			t.Log("get match message")
+			assert.NotNil(t, m)
+			c <- true
+		})
+		stream.OnChangeMessage(func(m *ChangeMessage) {
+			t.Log("get change message")
+			assert.NotNil(t, m)
+			c <- true
+		})
+		stream.OnActivateMessage(func(m *ActivateMessage) {
+			t.Log("get activate message")
+			assert.NotNil(t, m)
+			c <- true
+		})
+		err := stream.Connect(context.Background())
+		assert.NoError(t, err)
+	})
+	<-c
+}
+
+func TestLevel2(t *testing.T) {
+	c := make(chan string)
+	productIDs := []string{"BTC-USD"}
+
+	getSnapshot := false
+	getUpdate := false
+	t.Run("Run Level2", func(t *testing.T) {
+		stream := getTestStreamOrSkip(t)
+		for _, productID := range productIDs {
+			stream.Subscribe("level2", productID, types.SubscribeOptions{})
+		}
+		stream.OnOrderbookSnapshotMessage(func(m *OrderBookSnapshotMessage) {
+			t.Log("get orderbook snapshot message")
+			assert.NotNil(t, m)
+			c <- "snapshot"
+		})
+		stream.OnOrderbookUpdateMessage(func(m *OrderBookUpdateMessage) {
+			t.Log("get orderbook update message")
+			assert.NotNil(t, m)
+			c <- "update"
+		})
+		err := stream.Connect(context.Background())
+		assert.NoError(t, err)
+	})
+	for {
+		select {
+		case msg := <-c:
+			if msg == "snapshot" {
+				getSnapshot = true
+			} else if msg == "update" {
+				getUpdate = true
+			}
+		}
+		if getSnapshot && getUpdate {
+			break
+		}
+	}
+}
+
+func TestBalance(t *testing.T) {
+	accountsStr := os.Getenv("COINBASE_ACCOUNT_IDS")
+	accounts := strings.Split(accountsStr, ",")
+
+	c := make(chan bool)
+
+	t.Run("Run Balance", func(t *testing.T) {
+		stream := getTestStreamOrSkip(t)
+		for _, accountID := range accounts {
+			stream.Subscribe("balance", accountID, types.SubscribeOptions{})
+		}
+		triggered := false
+		stream.OnBalanceMessage(func(m *BalanceMessage) {
+			if triggered {
+				return
+			}
+			triggered = true
+			assert.NotNil(t, m)
+			t.Logf("get balance message: %v", *m)
+			c <- true
+		})
+		err := stream.Connect(context.Background())
+		assert.NoError(t, err)
+	})
+	<-c
+}
+
+func getTestStreamOrSkip(t *testing.T) *Stream {
+	if isCI, _ := strconv.ParseBool(os.Getenv("CI")); isCI {
```

Comment:
IntegrationTestWithPassphraseConfigured already checks the params for you, if the env vars are not defined, then the test will be skipped.

Reply:
Really?
I  check the `IntegrationTestWithPassphraseConfigured`.
It does not check the `"CI"` env variable.